monetio/sat/_tropomi_l2_no2_mm.py

```diff
@@ -103,9 +103,8 @@ def get_extra(varname_, *, dct_=None, default_group="PRODUCT"):
             # Tropopause pressure
             ptrop = xr.full_like(itrop, np.nan, dtype=ds["preslev"].dtype)
             for i in np.unique(itrop):
-                if np.isnan(i):
-                    continue
-                ptrop = xr.where(itrop == i, p.isel(z=int(i)), ptrop)
+                if i >= 0 and i < p.sizes["z"]:
+                    ptrop = xr.where(itrop == i, p.isel(z=int(i)), ptrop)
```

Comment:
How about this, to preserve the notion in the code that it can be NaN.
```suggestion
                if np.isnan(i) or i < 0 or i >= p.sizes["z"]:
                    continue
                ptrop = xr.where(itrop == i, p.isel(z=int(i)), ptrop)
```

Reply:
The ptrop array is set up initially filled with NaN and then replaces those with a pressure where there is a valid tropopause index to use. 
Both are giving the same result, so whichever syntax is preferred is fine.

Reply:
> The ptrop array is set up initially filled with NaN and then replaces those with a pressure where there is a valid tropopause index to use.

Yes, thanks, I wrote this part 😅

> whichever syntax is preferred is fine

I prefer preserving the notion that NaN can be one of the unique trop index values.

Reply:
Okay- there's just no actual difference in the output because NaN is getting replaced with NaN for a trop index of NaN.

Reply:
> no actual difference in the output

I know. Like I've said, I'd just prefer to preserve that NaN can be one of the unique trop index values.

>  NaN is getting replaced with NaN for a trop index of NaN

No it's not, it's being skipped since `p.isel(z=int(i))` would fail. `i >= 0 and i < p.sizes["z"]`, what you used, also guards against NaN `i`, it's just not explicit about it.

Reply:
> No it's not, it's being skipped since `p.isel(z=int(i))` would fail. `i >= 0 and i < p.sizes["z"]`, what you used, also guards against NaN `i`, it's just not explicit about it.

Whoops- I mis-read what was getting past the check and when the where gets evaluated. I'm not very familiar with using 'continue'.